src/core/Button/Button.tsx

```diff
@@ -41,102 +51,111 @@ export interface ButtonProps extends CompButtonProps, TokensProp {
    * Properties given to Icon-component
    */
   iconProps?: IconProps;
+  /** Event handler to execute when clicked
+   *  @default void
+   */
+  onClick?: (event: MouseEvent | KeyboardEvent) => void;
 }
 
+export interface ButtonProps extends InternalButtonProps, TokensProp {}
+
 const baseClassName = 'fi-button';
+const disabledClassName = `${baseClassName}--disabled`;
 const iconClassName = `${baseClassName}_icon`;
 const iconRightClassName = `${baseClassName}_icon--right`;
 const fullWidthClassName = `${baseClassName}--fullwidth`;
 
-const StyledButton = styled(
-  ({
-    tokens,
-    fullWidth,
-    variant = 'default',
-    className,
-    ...passProps
-  }: ButtonProps & { right?: boolean } & InternalTokensProp) => (
-    <CompButton
-      {...passProps}
-      className={classnames(className, {
-        [`${baseClassName}--${variant}`]: variant !== 'default',
-        [fullWidthClassName]: fullWidth,
-      })}
-    />
-  ),
-)`
-  ${(props) => baseStyles(props)}
-`;
-
-class ButtonWithIcon extends Component<ButtonProps & InternalTokensProp> {
+class BaseButton extends Component<ButtonProps> {
   render() {
     const {
+      fullWidth,
+      variant = 'default',
+      className,
+      disabled = false,
+      onClick,
+      'aria-disabled': ariaDisabled,
       icon,
       iconRight,
       iconProps = { className: undefined },
+      children,
       ...passProps
     } = this.props;
-    const { variant } = passProps;
     const { className: iconPropsClassName, ...passIconProps } = iconProps;
-
-    if (variant === 'unstyled') {
-      return <UnstyledButton {...passProps} />;
-    }
+    const onClickProps = !!disabled || !!ariaDisabled ? {} : { onClick };
```

Comment:
Maybe we could use singular, as there is only one property, --> `const onClickProp`

But not sure which is best, as this object is then spread later when used for button.

Reply:
Prop should be sufficient for this case I guess

Reply:
Changed.